XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs

```diff
@@ -29,5 +29,20 @@ public static bool IsValidArray(this Array array)
         {
             return array != null && array.Length > 0;
         }
+
+        /// <summary>
+        /// Extends an existing array to add a new item
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="array">The array to extend</param>
+        /// <param name="newItem">The item to add to the array</param>
+        /// <returns></returns>
+        public static T[] AddItem<T>(this T[] array, T newItem)
```

Comment:
```suggestion
        public static T[] AddItem<T>(this T[] array, T newItem, int insertAtIndex)
```

Reply:
Don't commit this. I just wanted to suggest an edit for later. After this PR is merged.